### ISSUE TITLE
Provide more context to what you're looking at in sidebar

### ISSUE THREAD

**Issue by msecret** [18F/cg-deck#382]
In order for deck user's to navigate the dashboard successfully without feeling confused or uncertain about what they're looking at we should implement some design features in the sidebar to reinforce the connection.

Acceptance criteria
- [x] To highlight current org and space in sidebar.
- [x] To hide all levels of hierarchy that you're not in.
- [x] Attempt to separate UI wording from user names through visual distinction.
- [x] Label organizations with UI label.

**Comment by thisisdano** [comment]
So, there are a couple issues here. Some we can/should address now, some may have to wait. 

1. Submenus outside the selected level should be hidden. (This could/should also include the top level. So, when you sign in and are prompted to select an org, all you see are the available orgs.)

<img width="818" alt="screen shot 2016-06-20 at 12 23 38 pm" src="https://cloud.githubusercontent.com/assets/11464021/16207776/d9885a46-36e3-11e6-9f7f-2333e6fa33a6.png">

2. The active level should be distinct from inactive levels in the hierarchy. (And perhaps also distinct from the hover state.)

<img width="581" alt="screen shot 2016-06-20 at 12 43 01 pm" src="https://cloud.githubusercontent.com/assets/11464021/16207935/94434cf6-36e4-11e6-9b17-d0851ae4f1d5.png">

**Followup:** Should `Spaces` and `Marketplace` be separate levels of hierarchy within an org?
  a. Can there be just a single markeplace? At the top/bottom like the new `Guide` button?
  b. Do spaces need to fall within a `Spaces` hed? Can they just be directly under the org name?
  c. Should a space's apps appear as selectable items in the sidebar?

**Comment by msecret** [comment]
- Marketplace does belong to an organization in the API, will probably keep for now.
- Should you have to cilck into spaces to see them?
- hierarchy is confusing, spaces is defined by color, the gray block is not super helpful.
- also issue on docs hierarchy sidebar.
- how much functionality are we dpulicating that we're also showing? We could show all the apps within the side bar as well but because we should the apps a different way within the space thing, it's redundant, but open question of whether it's easier in sidebar. 
- Give an explicit indication in sidebar into what level you're in.
- Hide all levels of hierarchy that aren't part of hierarchy you're in. Limit amount of extra data.
- Underlying issue here that's adding cognitive load, the names of apps spaces orgs and UI labels are not clearly distinguished from eachother. Use bold to distinguish UI labels from names of things.

**Comment by thisisdano** [comment]
Here's a version of how it could look with:
1) Colored active highlight colors for spaces and orgs.
2) Distinction for UI like `Overview` and `Marketplace`

![00-dash-sidebar 2x](https://cloud.githubusercontent.com/assets/11464021/16245502/0d3ca6bc-37b6-11e6-9a73-04069ad94d34.png)

**Comment by thisisdano** [comment]
Revision for clarity/consistency with boldface and expand/collapse icon.

![01-dash-sidebar 2x](https://cloud.githubusercontent.com/assets/11464021/16245964/03bdebd0-37b8-11e6-8fda-857e6d1f1b9d.png)

**Comment by msecret** [comment]
It makes sense to me. Maybe we should go for it and be sure to test with users later?

**Comment by thisisdano** [comment]
OK, I'll hack something together as best I can. I'll ask if I need help getting all the right hooks in the template.

**Comment by brittag** [comment]
I am still a little bit confused while looking at this, but it does seem better! I'll try to explain my confusion: "OK, it seems like yellow text means that the item is the active page, since overview is yellow and I'm on it, cool! Hm wait, why is 'spaces' yellow while I'm looking at an org - when the org seems active since it's highlighted by a bright background bar? Is there a difference in my status when a blue bar vs a yellow bar are highlighted, or do both mean the same kind of active page?"

**Comment by thisisdano** [comment]
Yeah, that makes sense — I may have bitten off more than necessary by going for an extra level of distinction between orgs and spaces. I kinda felt like that criticism was coming and warranted, so I'll probably just bring it back a notch and focus on the yellow as active indicator.

**Comment by thisisdano** [comment]
@brittag Does this feel like an improvement?
@msecret Does this still make sense?

I did a couple things:
- Changed hover indicator to be more related to active color
- Indented spaces
- Made active color consistent

Note: selecting an org (a parent element) will both toggle the sidebar between an open and closed state _and_ bring the user to the org's info panel. This means that the org sidebar may be open or closed while selected. I don't think this is a usability concern, however — the action is consistent: select an element, go to its info panel. If the element is a parent, toggle it between open and closed. I think this is preferable to an alternative where selecting the toggle selector (the little arrow) would be necessary to trigger a toggle. In this scenario, each parent element would then have a smaller toggle target within its primary target, which feels fussy and over-complicated.

Though feel free to convince me otherwise!

Also — clearly — this deviates from current sidebar functionality on the rest of the site. We could 
a) leave them different
b) try for a solution that works for both, even if it takes longer

![02-dash-sidebar 2x](https://cloud.githubusercontent.com/assets/11464021/16247821/84094dae-37c0-11e6-8903-21d395ef09d2.png)

**Comment by msecret** [comment]
What should the `cf-deck-testing` item be when you're on the `prod` space? and when you're on it's marketplace? For one you have it blue and for one you have it yetllow.

**Comment by thisisdano** [comment]
It would be blue in both instances. Blue would indicate an open (but inactive) org. The final example shows `cf-deck-testing` in a hover state.

**Comment by msecret** [comment]
Shouldn't `cf-deck-testing` be the same in the 2nd and 3rd examples? In both it's in a selected state. Or do you not want it selected when `prod` is?

**Comment by thisisdano** [comment]
**tl;dr:** Only one element will have the yellow bg at a time — the element whose info appears in the content panel.

**more:** The only element that should have a bold active yellow background should be the navigation element whose info is currently displayed in the content panel (or, if the info displayed in the content panel has no corresponding navigation element — like an app's info — its nearest relative, the space, would be active). So, if a space is active, it would have the yellow bg, but its parent container (the org) would not have the yellow bg. The parent element would be "open" (so you can see its spaces) and open parents are blue and unbold. 

Now, because apps do not appear in the sidebar, their nearest hierarchical relative in the sidebar would be active — their parent space. But, if apps _did_ appear in the sidebar, when an app's info appeared in the content panel then the app element itself would get the active yellow bg, not its parent space!

Does that make sense?

**Comment by brittag** [comment]
Overall this seems better to me; I could probably use this without feeling super lost. Yay!

It looks like Overview and Marketplace don't get a yellow bar when they're active - I'd probably go for the simpler approach of always yellow bar when something is active (even for those pages).

Leaving different seems fine for now, so we can iterate faster. We can ask people if they like this nav or the docs nav better.

**Comment by thisisdano** [comment]
Both above points seem reasonable to me.

**Comment by thisisdano** [comment]
OK, I've got a way forward for our immediate needs and I'll get to work on it. 

For fun, in the interim, let's flip some beans by looking ahead — beyond this issue to the role that sidebar navigation might play in future iterations of the dashboard. Can and should the sidebar be the primary navigation method for the dashboard at all? It can get complicated and busy in there — and the navigation inside often duplicates navigation information included in the content panel.

Hm!

You could imagine a version of the navigation where the sidebar was used only to get you to your favorite places (or, perhaps, to point you to places that might need your attention, like critical errors in apps...). This sidebar would act more like the sidebar in an OS — not like a directory tree, but loaded up with common destinations. The primary navigation would happen in the content area. 

It could look something like this:

![00-dash-sidebar-nonav 2x](https://cloud.githubusercontent.com/assets/11464021/16250147/02fb1638-37d0-11e6-89c2-efc5478e6ce6.png)

But that's enough of that for now, though. It's something we'll be talking about later!

**Comment by thisisdano** [comment]
And @brittag, here's the final mockup, including your last suggestion. So clear!

![03-dash-sidebar 2x](https://cloud.githubusercontent.com/assets/11464021/16250479/8dc7efa0-37d2-11e6-8026-0a23950849e3.png)

**Comment by brittag** [comment]
Yay, I like it! The bright yellow is also sort of cheery. :)

**Comment by msecret** [comment]
I attempted " To hide all levels of hierarchy that you're not in." and it makes the menu quite jumpy.

**Comment by brittag** [comment]
I think it'll be ok if we don't hide all the levels of hierarchy you're not in. I suspect the user need behind that was "this sidebar is confusing me about where I *am*, so I want to hide everything that's not relevant". This revised sidebar design will do a much better job of communicating where you are, so hiding might not be necessary.

**Comment by msecret** [comment]
@thisisdano can you accept this story? If it's not exactly as we wanted I think we can create another story for the changes.

**Comment by jeremiak** [comment]
Really nice work on this, all the acceptance criteria have been met and I'm going to close it for now. If we want to make further changes in the future, we can open a new issue then.
